clang/include/clang/Driver/Options.td

```diff
@@ -4028,6 +4028,13 @@ defm sycl_instrument_device_code
           BothFlags<[], [ClangOption, CLOption, DXCOption, CC1Option], " Instrumentation and Tracing "
                     "Technology (ITT) instrumentation intrinsics calls "
                     "(experimental)">>;
+defm sycl_decompose_functor
```

Comment:
Yes, but that only applies to NVPTX right now. However, I think I'm going to make this an opt-in until we are sure it doesn't create a performance regression.

Reply:
FYI, I made the behaviour opt-in here https://github.com/intel/llvm/pull/14434/commits/d7f7521b6d97d46878edda70d23ae1b7b63eed42 to avoid potential negative performance impact.